.travis/build_and_test.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```

Comment:
There's nothing wrong with this but if possible its better to decouple your scripts from TravisCI as much as possible.

Eventually builds will fail and you will want to reproduce this. If you have it so your build script doesn't depend on TravisCI specific environment variables (+ other TravisCI things) you'll have an easier time.

The way I typically do this is have `.travis.yml` call into a script that is TravisCI specific that sets up a bunch of environment variables and then calls into a script (that doesn't depend on TravisCI) that does the actual work.

The script that does the "actual work" is what you would use as an entry point on your local system for reproducing builds.